shopgate.php

```diff
@@ -74,845 +147,624 @@ public function __construct()
+
+		foreach ($sql as $query)
+			if ($query)
+				if (!$db::getInstance()->execute(trim($query)))
+					return false;
 
-		$this->log('INSTALLATION - database updates have been performed successfully', ShopgateLogger::LOGTYPE_DEBUG);
 		return true;
+
 	}
 
-	private function carrierCompatibility($carrier)
+	/**
+	 * uninstall
+	 *
+	 * @return bool
+	 */
+	public function uninstall()
 	{
-		// fix a bug in Prestashop before version 1.4.4.0 classes/cart.php function isCarrierInRange() range behavior
-		$this->log('INSTALLATION - checking carrier compatibility; getting price range', ShopgateLogger::LOGTYPE_DEBUG);
-		$rangePrices = RangePrice::getRanges($carrier->id);
-
-		if (empty($rangePrices))
-		{
-			$this->log('INSTALLATION - price range was empty, creating new one', ShopgateLogger::LOGTYPE_DEBUG);
-			$rangePrice = new RangePrice();
-			$rangePrice->id_carrier = $carrier->id;
-			$rangePrice->delimiter1 = 0.0;
-			$rangePrice->delimiter2 = 1000000.0;
-
-			$this->log('INSTALLATION - adding price range', ShopgateLogger::LOGTYPE_DEBUG);
-			if (!$rangePrice->add())
-			{
-				$this->log('installation failed: unable to add price range.', ShopgateLogger::LOGTYPE_ERROR);
-				return false;
-			}
-		}
-		else
-		{
-			$this->log('INSTALLATION - price range was found', ShopgateLogger::LOGTYPE_DEBUG);
-			$rangePrice = new RangeWeight($rangePrices[0]['id_range_price']);
-		}
-
-		$this->log('INSTALLATION - getting weight range', ShopgateLogger::LOGTYPE_DEBUG);
-		$rangeWeights = RangeWeight::getRanges($carrier->id);
-
-		if (empty($rangeWeights))
-		{
-			$this->log('INSTALLATION - weight range was empty, creating new one', ShopgateLogger::LOGTYPE_DEBUG);
-			$rangeWeight = new RangeWeight();
-			$rangeWeight->id_carrier = $carrier->id;
-			$rangeWeight->delimiter1 = 0.0;
-			$rangeWeight->delimiter2 = 1000000.0;
+		if (!parent::uninstall())
+			return false;
 
-			if (!$rangeWeight->add())
-			{
-				$this->log('installation failed: unable to weight price range.', ShopgateLogger::LOGTYPE_ERROR);
-				return false;
-			}
-		}
-		else
+		/**
+		 * remove carrier
+		 */
+		if ($shopgateCarrierId = Db::getInstance()->getValue('SELECT `id_carrier` FROM `'._DB_PREFIX_.'carrier` WHERE `name` = "Shopgate"'))
 		{
-			$this->log('INSTALLATION - weight range was found', ShopgateLogger::LOGTYPE_DEBUG);
-			$rangeWeight = new RangeWeight($rangeWeights[0]['id_range_weight']);
+			Db::getInstance()->Execute('DELETE FROM `'._DB_PREFIX_.'delivery` WHERE `id_carrier` = '.$shopgateCarrierId);
+			Db::getInstance()->Execute('DELETE FROM `'._DB_PREFIX_.'range_price` WHERE `id_carrier` = '.$shopgateCarrierId);
+			Db::getInstance()->Execute('DELETE FROM `'._DB_PREFIX_.'range_weight` WHERE `id_carrier` = '.$shopgateCarrierId);
+			Db::getInstance()->Execute('DELETE FROM `'._DB_PREFIX_.'carrier_lang` WHERE `id_carrier` = '.$shopgateCarrierId);
+			Db::getInstance()->Execute('DELETE FROM `'._DB_PREFIX_.'carrier_zone` WHERE `id_carrier` = '.$shopgateCarrierId);
```

Comment:
You should not remove lines from the database because of the relations between your existing data. However, you can set your carrier as deleted.